src/map/grid.c

```diff
@@ -96,6 +97,19 @@ int map_grid_direction_delta(int direction)
     }
 }
 
+int map_grid_chess_distance(int offset1, int offset2)
+{
+    int x1 = map_grid_offset_to_x(offset1);
+    int y1 = map_grid_offset_to_y(offset1);
+    int x2 = map_grid_offset_to_x(offset2);
+    int y2 = map_grid_offset_to_y(offset2);
+
+    int dx = abs(x2 - x1);
+    int dy = abs(y2 - y1);
+
+    return dx > dy ? dx : dy;
+}
+
```

Comment:
If I'm not mistaken, there's already a function that does this. `calc_distance`maybe? I'll check when I finish the review.